back/engines/commercial/custom_maps/spec/acceptance/map_layers_spec.rb

```diff
@@ -110,70 +114,107 @@
+            expect(attributes['title_multiloc']).to  eq title_multiloc
+            expect(attributes['layer_type']).to      eq 'geojson'
+            expect(attributes['geojson']).to         eq geojson
+            expect(attributes['url']).to             be_nil
+            expect(attributes['default_enabled']).to be true
+            expect(attributes['marker_svg_url']).to  eq marker_svg_url
+            expect(attributes['ordering']).to        eq ordering
+          end
+        end
+
+        context 'when passing both a geojson object and file' do
+          let(:geojson) { layer_attributes[:geojson] }
+
+          let(:geojson_file) do
+            {
+              base64: encode_json_file_as_base64('brussels-districts.geojson'),
+              filename: 'brussels-districts.geojson'
+            }
+          end
+
+          example_request 'Creates a map layer successfully using the geojson object' do
+            assert_status 200
+            expect(attributes['title_multiloc']).to  eq title_multiloc
+            expect(attributes['layer_type']).to      eq 'geojson'
+            expect(attributes['geojson']).to         eq geojson
+            expect(attributes['url']).to             be_nil
+            expect(attributes['default_enabled']).to be true
+            expect(attributes['marker_svg_url']).to  eq marker_svg_url
+            expect(attributes['ordering']).to        eq ordering
+          end
+        end
+
+        context 'when passing no geojson object or file' do
+          example_request 'Fails to create a map layer' do
+            assert_status 422
+          end
+        end
+      end
+
+      context 'when layer_type is esri_feature_service' do
+        let(:layer_type) { 'esri_feature_service' }
+
+        context 'when passing a url' do
+          let(:url) { 'https://some.domain.com/some_layer' }
+
+          example_request 'Creates a map layer successfully using a url' do
+            assert_status 200
+            expect(attributes['title_multiloc']).to  eq title_multiloc
+            expect(attributes['layer_type']).to      eq 'esri_feature_service'
+            expect(attributes['geojson']).to         be_nil
+            expect(attributes['url']).to             eq url
+            expect(attributes['default_enabled']).to be true
+            expect(attributes['marker_svg_url']).to  eq marker_svg_url
+            expect(attributes['ordering']).to        eq ordering
+          end
+        end
+
+        context 'when passing no url' do
+          example_request 'Fails to create a map layer' do
+            assert_status 422
+          end
         end
       end
     end
 
     patch 'web_api/v1/projects/:project_id/map_config/layers/:id' do
       with_options scope: :layer, required: true, with_example: true do
         parameter :title_multiloc,  'The name of the layer in multiple locales'
+        parameter :layer_type,      'The type of the layer (geojson or esri_feature_service)', required: true
```

Comment:
Do we want to support changing the layer type of an existing layer? Will this ever happen?

Reply:
Very good point 👍